src/rendering/SceneManager.cc

```diff
@@ -929,6 +937,117 @@ rendering::LightPtr SceneManager::CreateLight(Entity _id,
   return light;
 }
 
+/////////////////////////////////////////////////
+rendering::ParticleEmitterPtr SceneManager::CreateParticleEmitter(Entity _id,
+    const msgs::ParticleEmitter &_emitter, Entity _parentId)
+{
+  if (!this->dataPtr->scene)
+    return rendering::ParticleEmitterPtr();
+
+  if (this->dataPtr->particleEmitters.find(_id) !=
+     this->dataPtr->particleEmitters.end())
+  {
+    ignerr << "Particle emitter with Id: [" << _id << "] already exists in the "
+           <<" scene" << std::endl;
+    return rendering::ParticleEmitterPtr();
+  }
+
+  rendering::VisualPtr parent;
+  if (_parentId != this->dataPtr->worldId)
+  {
+    auto it = this->dataPtr->visuals.find(_parentId);
+    if (it == this->dataPtr->visuals.end())
+    {
+      // It is possible to get here if the model entity is created then
+      // removed in between render updates.
+      return rendering::ParticleEmitterPtr();
+    }
+    parent = it->second;
+  }
+
+  // Name.
+  std::string name = _emitter.name().empty() ? std::to_string(_id) :
+    _emitter.name();
+  if (parent)
+    name = parent->Name() +  "::" + name;
+
+  rendering::ParticleEmitterPtr emitter;
+  emitter = this->dataPtr->scene->CreateParticleEmitter(name);
+
+  // Type.
+  switch (_emitter.type())
```

Comment:
Before we run a switch on `_emitter.type()` to set the type for `emitter`, can we include a check to make sure `emitter` isn't `NULL`? 

(https://github.com/ignitionrobotics/ign-rendering/blob/cac86c83d488b1866374013a92e635ce604a1eee/include/ignition/rendering/Scene.hh#L941-L947)

Reply:
Done in defd33f6809b8b204ee91426bd48b6eb4c54fc7e